### PR TITLE
fix: Update SideNavigationLink component's prop type

### PR DESCRIPTION
**Pull request by aaryanporwal**
## Done

- [x] Update SideNavigationLink component's prop type to handle props properly.

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Steps for QA.

### Percy steps

- List any expected visual change in Percy, or write something like "No visual changes expected" if none is expected.

## Fixes

Fixes: # .

### CODE REVIEW

File: src/components/SideNavigation/SideNavigationLink/SideNavigationLink.tsx [hunk 1]
```diff
@@ -14,6 +14,10 @@ export type Props<L = LinkDefaultElement, E = HTMLAnchorElement> = Partial<
    * The navigation item's label.
    */
   label: ReactNode;
+  /**
+   * The navigation item's onClick event handler.
+   */
+  onClick?: () => void;
```
Comment (edlerd) [inline src/components/SideNavigation/SideNavigationLink/SideNavigationLink.tsx:20]: This looks wrong, Why add a prop, that is being ignored?
Should this be added to `SideNavigationBase` instead and also handled there?

Comment (aaryanporwal) [reply src/components/SideNavigation/SideNavigationLink/SideNavigationLink.tsx:20]: When I add forwardRef to the usage in anbox which has onClick prop. I get the error: `Property 'onClick' does not exist on type 'IntrinsicAttributes & Partial<Omit<Props<HTMLSpanElement>, "component">> & { label: ReactNode; component?: ElementType | ... 1 more ... | undefined; forwardRef?: Ref<...> | undefined; }'. Did you mean 'onclick'?`
image

Comment (aaryanporwal) [reply src/components/SideNavigation/SideNavigationLink/SideNavigationLink.tsx:20]: Should I add the onClick prop to SideNavBase?
Like this?

```
<Component ref={forwardRef} onClick={onClick} {...props}>
```

I don't get the above error when I add the onClick prop in SideNavBase.

Comment (huwshimi) [reply src/components/SideNavigation/SideNavigationLink/SideNavigationLink.tsx:20]: These props should get inferred from the component/element type so this suggests a problem with the way `Props` is defined. It looks like changing `Props` to this fixes the issue:

```typescript
export type Props<
  L = LinkDefaultElement,
  E = HTMLAnchorElement,
> = PropsWithSpread<
  {
    /**
     * The navigation item's label.
     */
    label: ReactNode;
    /**
     * The component or element to use for the link element e.g. `a` or `NavLink`.
     * @default a
     */
    component?: SideNavigationBaseProps<L>["component"];
    /**
     * A ref to pass to the link element.
     */
    forwardRef?: React.Ref<E> | null;
  },
  Omit<SideNavigationBaseProps<L>, "component">
>;
```

Then you can remove the `onClick` you added.

Comment (aaryanporwal) [reply src/components/SideNavigation/SideNavigationLink/SideNavigationLink.tsx]: Yes that's what I was suspecting. Thanks for the fix! It works now.

**Review by huwshimi — APPROVED** [review]
Looks good now, thanks for the improvement.

**Pull request merged**